spec/queries/homepage/articles_query_spec.rb

```diff
@@ -0,0 +1,101 @@
+require "rails_helper"
+
+RSpec.describe Homepage::ArticlesQuery, type: :query do
+  describe ".call" do
+    it "returns a relation object" do
+      expect(described_class.call).to be_a(ActiveRecord::Relation)
+    end
+
+    it "does not return draft articles" do
+      article = create(:article, published: false, published_at: nil)
+
+      expect(described_class.call.ids).not_to include(article.id)
+    end
+
+    it "returns both approved and unapproved articles by default" do
+      approved_article = create(:article, approved: true)
+      unapproved_article = create(:article, approved: false)
+
+      expected_result = [approved_article.id, unapproved_article.id]
+      expect(described_class.call.ids).to match_array(expected_result)
+    end
+
+    it "returns approved articles" do
+      article = create(:article, approved: true)
+
+      expect(described_class.call(approved: true).ids).to eq([article.id])
```

Comment:
Is this intended to exclude unapproved? If so, we may want to add an assertion for that.

Reply:
done